server/src/updaters/relationship-updaters.js

```diff
@@ -0,0 +1,202 @@
+        }
+
+        if (actorRelationship.status === relationshipStatuses.BLOCKED) {
+          throw new ServerError('user_already_blocked');
+        }
+      }
+
+      const [receiverRelationship] = getRelationshipsForReceiverUser(
+        relationships,
+        viewer.id,
+      );
+      if (receiverRelationship) {
+        if (isPendingFriendOrFriend(receiverRelationship.status)) {
+          const { user1, user2 } = receiverRelationship;
+          // cancels already received friend request or removes frienship then blocks user
+          await deleteRelationshipQuery(user1, user2);
+          await createRelationshipQuery(viewer.id, userID, status);
+        }
+
+        if (receiverRelationship.status === relationshipStatuses.BLOCKED) {
+          // users blocked each other
+          await createRelationshipQuery(viewer.id, userID, status);
+        }
+      }
+    }
+    // friend request acceptance
+  } else if (status === relationshipStatuses.FRIEND) {
+    const query = SQL`
+      SELECT user1, user2, status
+      FROM relationships
+      WHERE user1 = ${userID} AND user2 = ${viewer.id} AND status = ${relationshipStatuses.PENDING_FRIEND}
+    `;
+    const [relationships] = await dbQuery(query);
+    const [relationship] = relationships;
+
+    if (relationship) {
+      const { user1, user2 } = relationship;
+      await updateRelationshipQuery(user1, user2, relationshipStatuses.FRIEND);
+    } else {
+      throw new ServerError('friend_request_does_not_exist');
+    }
+  } else if (status === relationshipStatuses.KNOW_OF) {
+    invariant(false, `${status} is not currently supported`);
+  } else {
+    invariant(false, `invalid relationship status: ${status}`);
+  }
+}
+
+async function fetchRelationshipQuery(
+  firstUserID: string,
+  secondUserID: string,
+) {
+  const query = SQL`
+    SELECT user1, user2, status
+    FROM relationships
+    WHERE (user1 = ${firstUserID} AND user2 = ${secondUserID}) OR (user1 = ${secondUserID} AND user2 = ${firstUserID})
+  `;
+  return await dbQuery(query);
+}
+
+async function createRelationshipQuery(
+  firstUserID: string,
+  secondUserID: string,
+  status: RelationshipStatus,
+) {
+  const row = [firstUserID, secondUserID, status];
+  const query = SQL`
+    INSERT INTO relationships(user1, user2, status)
+    VALUES ${[row]}
+  `;
+  return await dbQuery(query);
+}
+
+async function deleteRelationshipQuery(
+  firstUserID: string,
+  secondUserID: string,
+) {
+  const query = SQL`
+    DELETE FROM relationships
+    WHERE user1 = ${firstUserID} and user2 = ${secondUserID}
+`;
+  return await dbQuery(query);
+}
+
+async function updateRelationshipQuery(
+  firstUserID: string,
+  secondUserID: string,
+  status: RelationshipStatus,
+) {
+  const query = SQL`
+    UPDATE relationships
+    SET status = ${status}
+    WHERE user1 = ${firstUserID} and user2 = ${secondUserID}
+  `;
+  return await dbQuery(query);
+}
+
+function getRelationshipsForActorUser(relationships, userID) {
```

Comment:
I'm not sure we need these functions. Instead of calling one function to fetch the data and another to process it, I think we should have just one function that fetches the data and processes it. It makes the code more readable and easy to understand